TCAT/Model/Route.swift

```diff
@@ -177,12 +177,32 @@ class Route: NSObject, JSONDecodable {
         
     }
     
-    // MARK: Process raw routes
+    // MARK: Process routes
     
     func isWalkingRoute() -> Bool {
         return directions.reduce(true) { $0 && $1.type == .walk }
     }
     
+    func getFirstDepartDirection() -> Direction? {
+        for direction in directions {
```

Comment:
small detail, but can use functional programming to return an optional in one line. 

`directions.first { $0.type == .depart }`

Change condition and ordering as needed.